src/mdacli/cli.py

```diff
@@ -271,147 +291,258 @@ def create_CLI(cli_parser, interface_name, parameters):
+        as coordinates. Alternatively, streams can be given.
+    topology_format : str, None
+        Provide the file format of the topology file; ``None`` guesses it from
+        the file extension. Can also pass a subclass of
+        :class:`MDAnalysis.topology.base.TopologyReaderBase` to define a custom
+        reader to be used on the topology file.
+    trajectory_format : str or list or object
+        provide the file format of the coordinate or trajectory file;
+        ``None`` guesses it from the file extension. Note that this
+        keyword has no effect if a list of file names is supplied because
+        the "chained" reader has to guess the file format for each
+        individual list member [``None``]. Can also pass a subclass of
+        :class:`MDAnalysis.coordinates.base.ProtoReader` to define a custom
+        reader to be used on the trajectory file.
+    atom_style : str
+        Customised LAMMPS `atom_style` information. Only works with
+        `topology_format = data`
+
+    Returns
+    -------
+    universe : `MDAnalysis.Universe`
+    """
+    universe = mda.Universe(topology,
+                            topology_format=topology_format,
+                            atom_style=atom_style)
 
-    ``kwargs`` contains all paramaters necessary for the analysis_callable
-    and the initlization of the MDAnalysis Universe.
+    if coordinates is not None:
+        universe.load_new(coordinates, format=trajectory_format)
+
+    return universe
+
+
+def run_analsis(analysis_callable,
+                universe_parameters,
+                mandatory_analysis_parameters,
+                optional_analysis_parameters=None,
+                run_parameters=None,
+                output_parameters=None):
+    """Perform main client logic.
 
     Parameters
     ----------
     analysis_callable : function
         Analysis class for which the analysis is performed.
+    mandatory_analysis_parameters : dict
+        Mandatory parameters for executing the analysis
+    optional_analysis_parameters : dict
+        Optional parameters for executing the analysis
+    run_parameters : dict
+        time frame parameters: start, stop, step, verbose
+    output_parameters : dict
+        output_prefix and output_directory
 
     Returns
     -------
     ac : `MDAnalysis.analysis.base.AnalysisBase`
         AnalysisBase instance of the given ``analysis_callable`` after run.
     """
-    verbose = kwargs.pop("verbose")
-    kwargs.pop("func")
+    if optional_analysis_parameters is None:
+        optional_analysis_parameters = {}
 
+    if run_parameters is None:
+        run_parameters = {}
+
+    if output_parameters is None:
+        output_parameters = {}
+
+    try:
+        verbose = run_parameters.pop("verbose")
+    except KeyError:
+        verbose = False
+
+    # Initilize Universe
     if verbose:
         print("Loading trajectory...", end="")
+    universe = create_universe(**universe_parameters)
```

Comment:
For another issue, we need to define a general logging library to configure debugging mode etc more nicely. But for now is fine.